tests/cases/FileUploadExtension.phpt

```diff
@@ -0,0 +1,208 @@
+<?php declare(strict_types = 1);
+
+namespace Tests\Cases;
+
+use Nette\DI\Compiler;
+use Nette\DI\InvalidConfigurationException;
+use Nette\InvalidStateException;
+use Nette\UnexpectedValueException;
+use Tester\Assert;
+use Tester\TestCase;
+use Tests\Fixtures\Filter\InvalidFilter;
+use Tests\Fixtures\Filter\ValidFilter;
+use Tests\Fixtures\Model\InvalidUploadModel;
+use Tests\Fixtures\Model\ValidUploadModel;
+use Tests\Fixtures\Renderer\InvalidRenderer;
+use Tests\Toolkit\Container;
+use Tests\Toolkit\Helpers;
+use Zet\FileUpload\Exception\InvalidArgumentException;
+use Zet\FileUpload\Filter\IMimeTypeFilter;
+use Zet\FileUpload\Model\IUploadModel;
+use Zet\FileUpload\Template\Renderer\Bootstrap4Renderer;
+use Zet\FileUpload\Template\Renderer\IUploadRenderer;
+
+require_once __DIR__ . '/../bootstrap.php';
+
+/**
+ * @testCase
+ */
+final class FileUploadExtension extends TestCase
+{
+
+	public function testConfig(): void
+	{
+		Assert::noError(function (): void {
+			Container::of()
+				->withDefaults()
+				->withCompiler(function (Compiler $compiler): void {
+					$compiler->addConfig(Helpers::neon('
+						fileUpload:
+							maxFiles: 10
+							maxFileSize: 2M
+							fileFilter: Zet\FileUpload\Filter\ImageFilter
+						'));
+				})
+				->build();
+		});
+
+		Assert::exception(function (): void {
+			Container::of()
+				->withDefaults()
+				->withCompiler(function (Compiler $compiler): void {
+					$compiler->addConfig(Helpers::neon('
+						fileUpload:
+							maxFiles: 10
+							maxFileSize: 2M
+							fileFilter: Zet\FileUpload\Filter\ImageFilter
+							uiMode: full
+						'));
+				})
+				->build();
+		}, InvalidConfigurationException::class, "Unexpected item 'fileUpload › uiMode'.");
+
+		Assert::exception(function (): void {
+			Container::of()
+				->withDefaults()
+				->withCompiler(function (Compiler $compiler): void {
+					$compiler->addConfig(Helpers::neon('
+						fileUpload:
+							maxFiles: null
+					'));
+				})
+				->build();
+		}, InvalidConfigurationException::class);
+
+		Assert::exception(function (): void {
+			Container::of()
+				->withDefaults()
+				->withCompiler(function (Compiler $compiler): void {
+					$compiler->addConfig(Helpers::neon('
+						fileUpload:
+							messages: hello
+					'));
+				})
+				->build();
+		}, InvalidConfigurationException::class, "The item 'fileUpload › messages' expects to be array, 'hello' given.");
+	}
+
+	public function testUploadModels(): void
+	{
+		$uploadModel = 'App\Model\MyUploadModel';
```

Comment:
Proc je tu tahle promenna? Neslo by to proste jen napsat naprimo?

Reply:
Napřímo by to možná šlo. Dal jsem to do proměnné kvůli hlášce na https://github.com/contributte/fileupload/blob/70fa3779371bd39a168f3fd30887f65119916d88/tests/cases/FileUploadExtension.phpt#L103.